solcore/optics/external_optics.py

```diff
@@ -4,21 +4,67 @@
 import types
 
 
+def calculate_absorption_rcwa(tmm_out):
```

Comment:
Since this is not specific for RCWA - I think - I would suggest you give it a more general name, as well as moving it to the optics_utilities.py we have discussed.

Reply:
Makes sense, I will update my branch with it.